src/main/scala/com/redis/RedisClient.scala

```diff
@@ -13,6 +15,16 @@ object RedisClient {
   case object SUM extends Aggregate
   case object MIN extends Aggregate
   case object MAX extends Aggregate
+
+  private def extractDatabaseNumber(connectionUri: java.net.URI) = {
+    Option(connectionUri.getPath).map(path =>
+      if (path.isEmpty) 0
+      else Try(Integer.parseInt(path.tail)) match {
+        case Success(dbNumber) => dbNumber
+        case Failure(e) => throw new IllegalArgumentException("Invalid database number", e)
+      })
+      .getOrElse(0)
+  }
 }
```

Comment:
Looks a bit over-engineered. Do we really need to construct the `Option` here ? I know it makes the whole thing more functional. But being a relatively low level library, I would like to avoid this. Rather ..

```
connectionUri.getPath match {
  case path if path.isEmpty => 0
  ...
}
```
Also you are throwing anyway if the `path` is invalid - so you can do away with the `Try` as well. WDYT ?